core/src/main/scala/kafka/tools/DumpLogSegments.scala

```diff
@@ -306,54 +305,66 @@ object DumpLogSegments {
                       isDeepIteration: Boolean,
                       maxMessageSize: Int,
                       parser: MessageParser[_, _]) {
-    val startOffset = file.getName().split("\\.")(0).toLong
+    val startOffset = file.getName.split("\\.")(0).toLong
     println("Starting offset: " + startOffset)
     val messageSet = FileRecords.open(file, false)
     var validBytes = 0L
-    var lastOffset = -1l
-    for (shallowLogEntry <- messageSet.shallowEntries(maxMessageSize).asScala) {
-      val itr = getIterator(shallowLogEntry, isDeepIteration)
-      for (deepLogEntry <- itr) {
-        val record = deepLogEntry.record()
-
-        if(lastOffset == -1)
-          lastOffset = deepLogEntry.offset
-        // If we are iterating uncompressed messages, offsets must be consecutive
-        else if (record.compressionType == CompressionType.NONE && deepLogEntry.offset != lastOffset +1) {
-          var nonConsecutivePairsSeq = nonConsecutivePairsForLogFilesMap.getOrElse(file.getAbsolutePath, List[(Long, Long)]())
-          nonConsecutivePairsSeq ::=(lastOffset, deepLogEntry.offset)
-          nonConsecutivePairsForLogFilesMap.put(file.getAbsolutePath, nonConsecutivePairsSeq)
-        }
-        lastOffset = deepLogEntry.offset
-
-        print("offset: " + deepLogEntry.offset + " position: " + validBytes +
-              " " + record.timestampType + ": " + record.timestamp + " isvalid: " + record.isValid +
-              " payloadsize: " + record.valueSize + " magic: " + record.magic +
-              " compresscodec: " + record.compressionType + " crc: " + record.checksum)
-        if (record.hasKey)
-          print(" keysize: " + record.keySize)
-        if (printContents) {
-          val (key, payload) = parser.parse(record)
-          key.foreach(key => print(s" key: $key"))
-          payload.foreach(payload => print(s" payload: $payload"))
+    var lastOffset = -1L
+    val entries = messageSet.batches(maxMessageSize).asScala
+    for (entry <- entries) { // this only does shallow iteration
+      if (isDeepIteration) {
+        for (record <- entry.asScala) {
+          if (lastOffset == -1)
+            lastOffset = record.offset
+          // If we are iterating uncompressed messages, offsets must be consecutive
+          else if (record.offset != lastOffset + 1) {
+            var nonConsecutivePairsSeq = nonConsecutivePairsForLogFilesMap.getOrElse(file.getAbsolutePath, List[(Long, Long)]())
+            nonConsecutivePairsSeq ::=(lastOffset, record.offset)
+            nonConsecutivePairsForLogFilesMap.put(file.getAbsolutePath, nonConsecutivePairsSeq)
+          }
+          lastOffset = record.offset
+
+          print("offset: " + record.offset + " position: " + validBytes +
+            " " + entry.timestampType + ": " + record.timestamp + " isvalid: " + record.isValid +
```

Comment:
Yeah... debated on this. All lower-case is hard to read (e.g. "partitionleaderepoch"), but I wasn't sure about compatibility.